grails-app/controllers/org/pih/warehouse/inventory/StockMovementController.groovy

```diff
@@ -283,6 +286,101 @@ class StockMovementController {
+        StockMovement stockMovement = stockMovementService.getStockMovement(params.id)
+        [stockMovement: stockMovement, comment: new Comment()]
+    }
+
+    def editComment = {
+        StockMovement stockMovement = stockMovementService.getStockMovement(params['stockMovement.id'])
+        if (!stockMovement) {
+            flash.message = "${g.message(code: 'default.not.found.message', args: [g.message(code: 'stockMovement.label', default: 'Stock Movement'), params['stockMovement.id']])}"
+            redirect(action: "list")
+            return
+        }
+        Comment comment = Comment.get(params?.id)
+        if (!comment) {
+            flash.message = "${g.message(code: 'default.not.found.message', args: [g.message(code: 'comment.label', default: 'Comment'), comment.id])}"
+            redirect(action: "show", id: stockMovement?.id)
+            return
+        }
+        if (comment.sender.id != session.user.id) {
+            flash.message = "${g.message(code: 'auth.notAuthorized.message')}"
+            redirect(action: "show", id: stockMovement?.id)
+            return
+        }
+        render(view: "addComment", model: [stockMovement: stockMovement, comment: comment])
+    }
+
+    def deleteComment = {
+        StockMovement stockMovement = stockMovementService.getStockMovement(params['stockMovement.id'])
+        if (!stockMovement) {
+            flash.message = "${g.message(code: 'default.not.found.message', args: [g.message(code: 'stockMovement.label', default: 'StockMovement'), params['stockMovement.id']])}"
+            redirect(action: "list")
+            return
+        }
+        Comment comment = Comment.get(params?.id)
+        if (!comment) {
+            flash.message = "${g.message(code: 'default.not.found.message', args: [g.message(code: 'comment.label', default: 'Comment'), params.id])}"
+            redirect(action: "show", id: stockMovement.id)
+            return
+        }
+
+        if (comment.sender.id != session.user.id) {
+            throw new UnsupportedOperationException("${g.message(code: 'errors.noPermissions.label')}")
+        }
+        stockMovementService.deleteComment(comment, stockMovement.requisition)
+        flash.message = "${g.message(code: 'default.comments.deleted.label')}"
+        redirect(action: "show", id: stockMovement.id)
+    }
+
+    def saveComment = {
+        StockMovement stockMovement = stockMovementService.getStockMovement(params['stockMovement.id'])
+        if (!stockMovement) {
+            flash.message = "${g.message(code: 'default.not.found.message', args: [g.message(code: 'stockMovement.label', default: 'StockMovement'), params['stockMovement.id']])}"
+            redirect(action: "list")
+            return
+        }
+        Comment comment = new Comment(params)
+        stockMovementService.saveComment(comment)
+        stockMovementService.addCommentToRequisition(comment, stockMovement.requisition)
+         if (comment?.hasErrors()) {
```

Comment:
shouldn't we revert the statement and do something like:
```groovy
if (comment.validate()) {
  stockMovementService.addCommentToRequisition(comment, stockMovement.requisition)
  flash.message = "${g.message(code: 'default.comments.created.label')}"
  redirect(action: "show", id: stockMovement.id)
  return
}
stockMovement = stockMovementService.getStockMovement(params['stockMovement.id'])
render(view: "addComment", model: [stockMovement: stockMovement, comment: comment])
```

Reply:
sure